src/extent.c

```diff
@@ -1930,6 +1980,11 @@ extent_merge_impl(tsdn_t *tsdn, arena_t *arena,
 		return true;
 	}
 
+	if (extent_dumpable_get(a) || extent_dumpable_get(b)) {
+		extent_ensure_dumpable(a);
+		extent_ensure_dumpable(b);
```

Comment:
<discussed offline>: we'll try making grow_retained its own state (which at any given point only 0 or 1 extents per arena can be in), and upgrade it to just retained before actually growing retained. This avoids some RSS explosion / VM fragmentation issues.